### PR TITLE
authenticate with Github using OAuth Token

### PR DESCRIPTION
**Pull request by abdulhannanali**

### CODE REVIEW

**Comment by abdulhannanali** [comment]
@roonyh let me pull some updates regarding 
https://github.com/roonyh/fossasia-gci-companion/issues/3#issuecomment-169388555

**Comment by abdulhannanali** [comment]
@roonyh you can try it just don't forget to create a new application in case you haven't 
https://github.com/settings/applications/new

and replace the client_id
and client_secret with their associated values :)

**Comment by abdulhannanali** [comment]
:smile: :smile: :smile:

**Comment by abdulhannanali** [comment]
@roonyh do we need to add a login button for github, or should it be performed automatically? I can add one. It will take  a little more code complexity though in terms of talking more between ipc render and ipc main processes.

**Comment by roonyh** [comment]
@abdulhannanali yes. Please add a button. Yes communicate with the communicate between renderer and main using ipc. And can we use a webview tag to display login from the same window?

**Comment by abdulhannanali** [comment]
@roonyh I guess we need a different window for github login.

**Comment by roonyh** [comment]
@abdulhannanali We'll get to that later. For now add the button. I tried out the PR. Worked pretty nice!

**Comment by abdulhannanali** [comment]
@roonyh where should the github button go? I am bad at layouts

**Comment by abdulhannanali** [comment]
@roonyh i have designed the events, can we make a new issue for the button, it's going to be changed any ways when the views are modularized?

**Comment by roonyh** [comment]
Pretty sure I am worse at layouts :)

Just add it for now. Somewhere on the top. It does not have to look nice at all.

**Comment by abdulhannanali** [comment]
@roonyh a basic button has been added for test purposes. tell me if it's okay.

File: index.html [hunk 1]
```diff
@@ -46,7 +49,5 @@ <h2 class="mdl-card__title-text">Your Global Git Config</h2>
     </main>
   </div>
 </body>
-<script>
-require("./git-config.js");
-</script>
```
Comment (roonyh) [inline index.html:51]: Don't remove this. Or call it inside index.js

**Comment by roonyh** [comment]
Thanks much! This is very nice! See the line note.

File: index.html (hunk 1 above)
Comment (abdulhannanali) [inline index.html:51]: okay! sorry! I forgot to add it.

Comment (roonyh) [inline index.html:51]: np. add it and mark your gci-task to review.

**Comment by abdulhannanali** [comment]
@roonyh Thank you! I have required the git-config. . Since electron is very fun and we are also using git in it. I will make sure to contribute and keep working on it in the future :smile:

**Comment by roonyh** [comment]
@abdulhannanali great!! :+1:

**Comment by abdulhannanali** [comment]
@roonyh please tell if there are more updates to be made before mergin :smile:

**Pull request merged**